src/ImageSharp/Memory/Allocators/SimpleGcMemoryAllocator.cs

```diff
@@ -19,6 +20,13 @@ public override IMemoryOwner<T> Allocate<T>(int length, AllocationOptions option
     {
         Guard.MustBeGreaterThanOrEqualTo(length, 0, nameof(length));
 
+        int lengthInBytes = length * Unsafe.SizeOf<T>();
```

Comment:
For V4 I'd like to have this in `MemoryAllocator` so we don't have to repeat the check.